### ISSUE TITLE
Multi-sig wallets

### ISSUE THREAD

**Issue by CyrusVorwald** [icon-project/icon-bridge#155]
Number of signers to verify a transaction is a known attack vector that has compromised other bridges. We need to avoid having a single point of failure. This means the private keys should be held by different parties in different servers.

For reference, Wormhole currently has 19 guardians and requires 2/3 of all validator signatures to approve a VAA.

**Comment by FezBox** [comment]
@CyrusVorwald-ICON is this still a valid Epic?

**Comment by CyrusVorwald-ICON** [comment]
Yes. This is of course very important and should become a priority pretty soon
